cabal-install/tests/UnitTests/Distribution/Client/Init/NonInteractive.hs

```diff
@@ -1265,3 +1271,35 @@ testGo label f g h inputs = testCase label $
   case (_runPrompt $ f emptyFlags) (NEL.fromList inputs) of
     Left  x -> g x
     Right x -> h x
+
+cliParserTests :: TestTree
+cliParserTests = testGroup "cli parser"
+  [ testCase "Parse single dependency" $ do
+      flags <- assumeAllParse $ commandParseArgs initCommand False ["-d", "base"]
+      flags @?= dummyFlags
+        { dependencies = fmap mkStringyDep ["base"]
+        }
+  , testCase "Parse multiple dependency flags" $ do
+      flags <- assumeAllParse $ commandParseArgs initCommand False
+        ["-d", "base", "-d", "vector"]
+      flags @?= dummyFlags
+        { dependencies = fmap mkStringyDep ["base", "vector"]
+        }
+  , testCase "Parse list of dependencies" $ do
+      flags <- assumeAllParse $ commandParseArgs initCommand False
+        ["-d", "base,vector"]
+      flags @?= dummyFlags
+        { dependencies = fmap mkStringyDep ["base", "vector"]
+        }
+  , testCase "Fail to parse multiple exposed-modules" $ do
+      flags <- assumeAllParse $ commandParseArgs initCommand False
+        ["-o", "Test", "-o", "Test2"]
+      flags @?= dummyFlags
+        { exposedModules = Flag (fmap fromString ["Test2"])
+        }
```

Comment:
This test-case proves the same problem for all other occurrences of `Flag [a]` in `InitFlags`.